quickwit/quickwit-proto/src/control_plane/mod.rs

```diff
@@ -52,22 +52,22 @@ impl From<ControlPlaneError> for MetastoreError {
 
 impl From<ControlPlaneError> for tonic::Status {
     fn from(error: ControlPlaneError) -> Self {
-        match error {
-            ControlPlaneError::Internal(message) => tonic::Status::internal(message),
-            ControlPlaneError::Metastore(error) => error.into(),
-            ControlPlaneError::Unavailable(message) => tonic::Status::unavailable(message),
-        }
+        let grpc_status_code = error.error_code().to_grpc_status_code();
+        let error_json =
+            serde_json::to_string(&error).expect("control plane error should be JSON serializable");
+
+        tonic::Status::new(grpc_status_code, error_json)
     }
 }
 
 impl From<tonic::Status> for ControlPlaneError {
     fn from(status: tonic::Status) -> Self {
-        match status.code() {
-            tonic::Code::Unavailable => {
-                ControlPlaneError::Unavailable(status.message().to_string())
-            }
-            _ => ControlPlaneError::Internal(status.message().to_string()),
-        }
+        serde_json::from_str(status.message()).unwrap_or_else(|_| {
```

Comment:
JSON serialization of errors is a bit brittle, but it'll do for now. We need a way to go through the gRPC layer without losing the error variants.